packages/wrangler/src/module-collection.ts

```diff
@@ -97,12 +103,44 @@ export default function makeModuleCollector(props: {
 
           rule.globs.forEach((glob) => {
             build.onResolve(
-              { filter: globToRegExp(glob) },
+              {
+                // we create a regexp that can detect any of the module rule
+                // globs, as well as any of the legacy module references
+                filter: new RegExp(
```

Comment:
this logic is actually wrong, making this a draft while I figure out a better solution

Reply:
ok fixed now.